### PR TITLE
Hts 534 update eligibility check interface again

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

**Comment by AndyHWChung** [comment]
https://drive.google.com/drive/folders/0B-uLGjQZTHTzZXFXNHA2aXJQenM

File: test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala [hunk 1]
```diff
@@ -42,8 +42,10 @@ class EligibilityCheckConnectorSpec extends TestSupport with GeneratorDrivenProp
 
   implicit val resultArb: Arbitrary[EligibilityCheckResult] = Arbitrary(for {
     result ← Gen.alphaStr
+    resultCode ← Gen.choose(1, 10)
     reason ← Gen.alphaStr
-  } yield EligibilityCheckResult(result, reason))
+    reasonCode ← Gen.choose(1, 10)
+  } yield EligibilityCheckResult(result, resultCode, reason, reasonCode))
```
Comment (sureshhmrc) [inline test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala]: the yield wont produce a valid `EligibilityCheckResult` entity here as its generating random strings and ints :)

**Comment by sureshhmrc** [comment]
ignore my comment - its been merged already :)

File: test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala (hunk 1 above)
Comment (AndyHWChung) [reply test/uk/gov/hmrc/helptosave/connectors/EligibilityCheckConnectorSpec.scala]: in the backend we don't particularly care if it's valid or not - the validation is done in the frontend. The backend is basically acting as a dumb proxy to DES for the frontend

**Pull request merged**